src/app/models/Contests.php

```diff
@@ -550,4 +544,38 @@ public function cleanJuriesPermisions()
             $jury->unsetJuryPermission();
         }
     }
+
+    public function setVeredictToPublished() : bool
+    {
+        $veredict = $this->getVeredict(); 
+        if (!$veredict){
+            return false;
+        }
+        return $veredict->changePublishedStatus();      
+    }
+
+    protected function getVeredict() : ?VeredictContestAttachedFile
+    {
+        $veredictClassName = get_class(new VeredictContestAttachedFile());
+        foreach ($this->attachedFiles as $file) {
+            if(get_class($file) === $veredictClassName){
+                return $file;
+            }
+        }
+
+        return null;
+    }
+
+    public function getApprovalResolution() : ?ContestAttachedFile
+    {
+        foreach ($this->attachedFiles as $file) {
+            if($file->documentType->code === DocumentType::APPROVAL_RESOLUTION_CONTEST
+                ||
+                $file->documentType->code === DocumentType::APPROVAL_RESOLUTION_CONTEST_EVALUATION_COMMISSION){
+                    return $file;
+            }
+        }
```

Comment:
Qué tal si haces un query sobre `attachedFiles()` que tenga un filtro `where in` donde obtenga alguno que posea el tipo buscado?